utils/s2n_mem.c

```diff
@@ -88,15 +87,16 @@ int s2n_realloc(struct s2n_blob *b, uint32_t size)
         GUARD(s2n_free(b));
         S2N_ERROR(S2N_ERR_MLOCK);
     }
-    b->mlocked = 1;
 
     return 0;
 }
 
+#include<stdio.h>
+
 int s2n_free(struct s2n_blob *b)
 {
     int munlock_rc = 0;
-    if (b->mlocked) {
+    if (use_mlock && b->data) {
```

Comment:
That field is currently unreliable because of how we free connections/configs: 
https://github.com/awslabs/s2n/blob/master/tls/s2n_connection.c#L140
https://github.com/awslabs/s2n/blob/master/tls/s2n_config.c#L253

I suppose we could set b->allocated on that temporary blob because know that s2n_{config, connection}_new used s2n_alloc